cli_tools/gce_windows_upgrade/upgrader/workflows.go

```diff
@@ -0,0 +1,216 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package upgrader
+
+import (
+	"context"
+	"fmt"
+
+	daisyutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/daisycommon"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	"google.golang.org/api/compute/v1"
+)
+
+func prepare(ctx context.Context, preparationVarMap map[string]string, params *UpgradeParams) (*daisy.Workflow, error) {
+	// 'windows-startup-script-url' exists, backup it
+	if windowsStartupScriptURLBackup != nil {
+		fmt.Printf("\nDetected current '%v', value='%v'. Will backup to '%v'.\n\n", metadataKeyWindowsStartupScriptURL,
+			*windowsStartupScriptURLBackup, metadataKeyWindowsStartupScriptURLBackup)
+		preparationVarMap["original_startup_script_url"] = *windowsStartupScriptURLBackup
+	}
+	prepWf, err := daisycommon.ParseWorkflow(upgradePreparationWorkflowPath, preparationVarMap,
+		params.project, params.zone, params.ScratchBucketGcsPath, params.Oauth, params.Timeout,
+		params.Ce, params.GcsLogsDisabled, params.CloudLogsDisabled, params.StdoutLogsDisabled)
+	if err != nil {
+		return nil, err
+	}
+	if windowsStartupScriptURLBackup == nil {
+		if !windowsStartupScriptURLBackupExists {
+			fmt.Printf("\nNo existing '%v' detected. Won't backup it.\n\n", metadataKeyWindowsStartupScriptURL)
```

Comment:
What should this message be about? I would have expected something like "No existing startup script '...' found"?

Reply:
Changed